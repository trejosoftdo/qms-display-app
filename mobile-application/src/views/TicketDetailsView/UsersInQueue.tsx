import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { StyleSheet } from 'react-native';
import { Text } from 'react-native-paper';
import { TRANSLATION_USERS_IN_QUEUE_KEY } from '../../common/translations/translation-keys';


/**
 * UsersInQueueProps defines the props for the Users In Queue Component.
 *
 * @property total - total of users in queue
 */
interface UsersInQueueProps {
  total: number;
};

/**
 * A component to display the number of users who are in queue
 *
 * @param {UsersInQueueProps} props - The props for the Users In Queue component.
 */
const UsersInQueue: React.FC<UsersInQueueProps> = (props: UsersInQueueProps) => {
  const { t } = useTranslation();
  return (
    <Text style={styles.container} variant="titleMedium">
      {t(TRANSLATION_USERS_IN_QUEUE_KEY, { total: props.total })}
    </Text>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingBottom: 32,
  },
});

export default UsersInQueue;
