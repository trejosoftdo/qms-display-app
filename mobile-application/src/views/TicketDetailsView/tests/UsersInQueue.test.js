import React from 'react';
import UsersInQueue from '../UsersInQueue';


describe('UsersInQueue component', () => {
  const mockProps = {
    total: 16,
  };

  afterEach(() => {
    jest.clearAllMocks();
  })

  it('renders the component as expected with the given props', () => {
    expect(renderShallow(<UsersInQueue {...mockProps} />)).toMatchSnapshot();
  });
});
