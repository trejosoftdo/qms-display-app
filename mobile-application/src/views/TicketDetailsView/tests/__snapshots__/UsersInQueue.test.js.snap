// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersInQueue component renders the component as expected with the given props 1`] = `
<ForwardRef(Text)
  style={
    {
      "paddingBottom": 32,
    }
  }
  variant="titleMedium"
>
  Translated[translation:usersInQueue]({"total":16})
</ForwardRef(Text)>
`;
